### DISCUSSION TITLE
How do I make yank, delete, change, etc not copy to system clipboard?

### DISCUSSION THREAD

**Discussion by jmbeach** [LazyVim/LazyVim#2501] [Q&A] [answered]
Hello, when I yank something, change something, delete something, etc, it seems to not only copy to the default buffer, but the actual clipboard as well.
I would like to disable this.
Is there a specific setting I can change?
To be clear, I still want to be able to run `"+y` to yank to the clipboard. But I *only* want that to yank to the clipboard.

**Comment by mikalcallahan** [comment]
This isn't specific to LazyVim but a vim option. Should be able to do `:set clipboard=` - See [stackoverflow](https://stackoverflow.com/questions/38617304/how-to-disable-vim-pasting-to-from-system-clipboard)

**Comment by cerebrl** [comment, marked as answer]
There's an issue here that has a discussion about this topic: https://github.com/LazyVim/LazyVim/issues/2327

**Reply by jmbeach** [reply]
Thanks. I am doing `vim.opt.clipboard = ""` and thought that wasn't working because even though yank etc weren't copying to clipboard, `"+y` wasn't yanking to clipboard either so I thought I completely broke the clipboard integration.

Now it seems to be working completely fine, so I'm not sure exactly what was going on. To be fair, I'm trying to do neovim, within wsl so clipboard integration might be inherently wonky

**Reply by jmbeach** [reply]
It was a wsl thing. This fixed my integration with the keyboard. Adding just in case it helps anyone else:

```lua
-- Use the right clipboard when on WSL
local function is_wsl()
  local wsl_check = os.getenv("WSL_DISTRO_NAME") ~= nil
  return wsl_check
end

-- Install win32yank with cargo and then run:
-- sudo ln -s /mnt/c/Users/jared/.cargo/bin/win32yank.exe /usr/local/bin/win32yank
if is_wsl() then
  vim.g.clipboard = {

    name = "win32yank-wsl",
    copy = {
      ["+"] = "win32yank.exe -i --crlf",
      ["*"] = "win32yank.exe -i --crlf",
    },

    paste = {
      ["+"] = "win32yank.exe -o --lf",
      ["*"] = "win32yank.exe -o --lf",
    },
    cache_enabled = 1,
  }
end
```

**Reply by jacksonvieiracs** [reply]
Thanks help me a lot, I think is not necessary `.exe` extension in the command configuration

**Comment by Flimm** [comment]
Open `~/.config/nvim/lua/config/options.lua` (if using LazyVim there) and add this line:

     vim.opt.clipboard = ''
